### DISCUSSION TITLE
A GUI for beets

### DISCUSSION THREAD

**Discussion by pSpitzner** [beetbox/beets#5353] [Show and tell]
Hi all,

@semohr and I have built a gui for beets and we would like to get some feedback.
There is still lots to do, but for a proof-of-concept we think its a good time to get the community on board.

Because our main use case is a home-server that hosts the music collection, we based this on a docker container with a web-frontend. This is what it currently looks like:

https://github.com/user-attachments/assets/05994008-0d9d-463b-acc0-6cb96e881bd4

For details and setup instructions, have a look at the repo:

[pSpitzner/beets-flask](https://github.com/pSpitzner/beets-flask)

Any input is greatly appreciated!
Cheers

**Comment by pSpitzner** [comment]
**short update and shameless bump**

We just pushed the update to `v0.1.0`:
- much easier onboarding and improved readme
- now on docker hub
- interactive (live) import
- auto import for good matches

If some of the beets backend pros are stumbling across this:
- We added a bunch of type hints (e.g. for pipeline mutator stages)
- abstractions around music types (album, track, file info)
- stateful session and tweaked pipeline stages
- and we are wondering which parts of this would make a good PR

Feel free to reach out!

**Comment by JOJ0** [comment]
This is crazy cool! I'm not sure if I have a personal use case but I find especially cool how you wrap cli stuff, eg imports with selection possibilities over a gui! Magic?

**Reply by semohr** [reply]
It's some magic, but also lots of duct tape :grin: 

We extended the beets importer session and added some functionality for serialization and parallel processing. If you are interested you may find the source for our base session [here](https://github.com/pSpitzner/beets-flask/blob/96360da09eca65cd2cd16400890a3c7d6c483932/backend/beets_flask/beets_sessions.py#L37) and the interactive session [here](https://github.com/pSpitzner/beets-flask/blob/96360da09eca65cd2cd16400890a3c7d6c483932/backend/beets_flask/importer/session.py#L17) which allows processing matches simultaneously.

**Reply by pSpitzner** [reply]
Thanks for the praise @JOJ0,
appreciate it :)

I have been wondering for a while on how to keep track of where beets is going, whats coming next, and well, how we can add things we might need for the frontend.
Currently we overload/inherit where possible, but there was the temptation to monkey patch here and there.

Are github discussions the only place, or do the maintainers coordinate elsewhere, too?
